backend/src/main/scala/cromwell/backend/BackendActor.scala

```diff
@@ -0,0 +1,146 @@
+package cromwell.backend
+
+import akka.actor.{Actor, ActorLogging}
+import akka.event.LoggingReceive
+import cromwell.backend.BackendActor._
+import cromwell.backend.model.{JobDescriptor, WorkflowDescriptor}
+
+import scala.concurrent.Future
+import scala.util.{Success, Failure}
+
+object BackendActor {
+
+  // Commands
+  sealed trait BackendActorMessage
+  case class Execute(jobDescriptor: JobDescriptor) extends BackendActorMessage
+  case class Abort(jobDescriptor: JobDescriptor) extends BackendActorMessage
+  case object AbortAll extends BackendActorMessage
+  case class Recover(jobDescriptor: JobDescriptor) extends BackendActorMessage
+
+  // Events
+  sealed trait BackendActorEvent
+
+  sealed trait ExecutionEvent extends BackendActorEvent
+  case class ExecutionSucceeded(result: String) extends ExecutionEvent //Result TBD.
+  case class ExecutionFailed(throwable: Throwable) extends ExecutionEvent
+
+  sealed trait AbortEvent extends BackendActorEvent
+  case class AbortSucceeded(jobDescriptor: JobDescriptor) extends AbortEvent
+  case class AbortFailed(throwable: Throwable) extends AbortEvent
+
+  sealed trait AbortAllEvent extends BackendActorEvent
+  case class AbortAllSucceeded(jobDescriptor: List[JobDescriptor]) extends AbortAllEvent //Attributes TBD.
+  case class AbortAllFailed(throwable: Throwable) extends AbortAllEvent
+
+  sealed trait RecoverEvent extends BackendActorEvent
+  case class RecoverSucceeded(jobDescriptor: JobDescriptor) extends RecoverEvent //Same happens here.
+  case class RecoverFailed(throwable: Throwable) extends RecoverEvent
+
+  sealed trait ValidationEvent extends BackendActorEvent
+  case object ValidationSucceeded extends ValidationEvent
+  case class ValidationFailed(throwable: Throwable) extends ValidationEvent
```

Comment:
We *might* be able to do away with our up front fast validation business requirement which simplifies this